rfc-126-custom-configuration-for-dependabot.md

```diff
@@ -0,0 +1,246 @@
+
+#### Limit updates to lockfile
+
+Another option might be to limit Dependabot to only raise PRs against the `Gemfile.lock` file. The lockfile is used to pin a specific version, whereas the `Gemfile` can contain a wider version range (for example `~> 1.0` means `1.0 all the way up to but not including 2.0`).
+
+Unfortunately applying this would mean [we lose the security updates that do require an update to the `Gemfile`][dependabot-security-lockfile].
+
+[dependabot-security-lockfile]: https://github.com/dependabot/feedback/issues/937
+
+#### Major/minor/patch
+
+This solution proposed above ignores all updates for libraries not in one of the three categories, regardless of whether the update is a major, minor or patch update. An alternative might be to include PRs for major updates for other libraries.
+
+Looking through the Dependabot config docs, [it doesn't seem to be possible to enable major updates][dependabot-allowed-updates]. Regardless, major updates tend to be the ones that take the longest time to merge, so enabling them wouldn't reduce much of the workload.
+
+[dependabot-allowed-updates]: https://dependabot.com/docs/config-file/#allowed_updates
+
+#### Opt-out rather than opt-in
+
+Rather than having a list of libraries we allow updating, we could instead [maintain a list of libraries we ignore updates for][dependabot-ignored-updates].
+
+This would allow us to remove the hardest PRs from updates, but it would require us to be more on top of maintaining the list. I'm also not sure how it would work with enabling security updates regardless.
+
+[dependabot-ignored-updates]: https://dependabot.com/docs/config-file/#ignored_updates
+
+## Consequences and measuring success
+
+Using [the statistics from above](#Statistics), we can make some estimates on what affect this would have had if it had been enabled from the beginning:
+
+- 7378 PRs wouldn't have been raised (10216 rather than the 17526 we've had so far).
+- The mean time to merge would have decreased from almost 3 days to just under 2 days.
+- The ignored libraries take on average 3.5 days to merge.
+
+In theory, by limiting the number of libraries that get updated by Dependabot, our applications would end up less up to date. However, by prioritising certain libraries over others, we should end up in a situation where our applications are actually _more_ up to date in the areas we care about (security, frameworks, internal libraries).
+
+We should expect to see the average time for PRs to be merged in (and in particular security PRs) to reduce. We should also expect to see a reduction in the number of open PRs, and also a reduction in the number of deployments.
+
+A less measurable metric is that we would expect developers to be spending less time fixing issues in Dependabot PRs.
+
+If we see a situation where most days we don't have any new Dependabot PRs, we could think about losening these restrictions and opening up the number of libraries we want to received updates on.
```

Comment:
What do you think about some kind of review of the situation in X months?

Reply:
That sounds like a good idea to me! I've added a sentence about this in 32511dd.